src/main/kotlin/org/jetbrains/intellij/dependency/MavenPluginsRepository.kt

```diff
@@ -1,41 +1,55 @@
 package org.jetbrains.intellij.dependency
 
+import groovy.lang.Closure
 import groovy.transform.CompileStatic
 import org.gradle.api.Project
+import org.gradle.util.ConfigureUtil
 import org.jetbrains.intellij.debug
 import java.io.File
 import java.net.URI
 
 @CompileStatic
-class MavenPluginsRepository(private val repositoryUrl: String) : PluginsRepository {
+class MavenPluginsRepository(private val repositoryUrl: String?, private val mavenClosure: Closure<Any>? = null) :
+    PluginsRepository {
 
     private var resolvedDependency = false
 
     override fun resolve(project: Project, plugin: PluginDependencyNotation): File? {
         val dependency = plugin.toDependency(project)
+        val repository = project.repositories.maven(ConfigureUtil.configureUsing(mavenClosure))
 
-        debug(project.name, "Adding Maven repository to download $dependency - $repositoryUrl")
-        val mavenRepository = project.repositories.maven { it.url = URI.create(repositoryUrl) }
+        debug(project.name, "Adding Maven repository to download $dependency - ${repositoryUrl ?: repository.url}")
+        val mavenRepository = if (repositoryUrl == null) {
+            repository
+        } else {
+            project.repositories.maven { it.url = URI.create(repositoryUrl) }
+        }
 
         var pluginFile: File? = null
         try {
             val configuration = project.configurations.detachedConfiguration(dependency)
             pluginFile = configuration.singleFile
             resolvedDependency = true
         } catch (e: Exception) {
-            debug(project.name, "Couldn't find $dependency in $repositoryUrl", e)
+            debug(project.name, "Couldn't find $dependency in ${repositoryUrl ?: repository.url}", e)
```

Comment:
`project.name` context is ignored, use `project`

Reply:
@zolotov not anymore, to make it compatible with Configuration Cache, I'm avoiding passion the whole `Project`/`Task` context to the logger – just the `String`.

Reply:
well, I noticed :/

![image](https://user-images.githubusercontent.com/140920/116864503-3abb4780-ac08-11eb-999c-ae0821129a9e.png)

Reply:
please consider this one as resolved then